src/multicellular_stage/MulticellularSpecies.cs

```diff
@@ -188,6 +199,54 @@ public override float GetPredationTargetSizeFactor()
         return totalOrganelles;
     }
 
+    public void SetupWorldEntities(IWorldSimulation worldSimulation)
+    {
+        ((MicrobeVisualOnlySimulation)worldSimulation).CreateVisualisationColony(this);
+    }
+
+    public bool StateHasStabilized(IWorldSimulation worldSimulation)
+    {
+        return true;
+    }
+
+    public Vector3 CalculatePhotographDistance(IWorldSimulation worldSimulation)
+    {
+        float radius = 0.0f;
+
+        Vector3 center = Vector3.Zero;
+
+        int count = Cells.Count;
+        for (int i = 0; i < count; ++i)
+        {
+            center += Hex.AxialToCartesian(Cells[i].Position);
+        }
+
+        center /= count;
+
+        foreach (var entity in worldSimulation.EntitySystem)
+        {
+            if (!entity.Has<CellProperties>())
+                continue;
+
+            ref var cellProperties = ref entity.Get<CellProperties>();
+
+            // This uses the membrane as radius is not set as the physics system doesn't run
+            if (!cellProperties.IsMembraneReady())
+                throw new InvalidOperationException("Microbe doesn't have a ready membrane");
```

Comment:
Shouldn't this be checked in `StateHasStabilized` as this truly may not be ready as when the membranes are calculated is kind of performance dependent?

Reply:
Though I guess the microbe simulation already has code assuming that. For putting similar code in a similar spot, I think this distance calculation would make sense to be in the MicrobeVisualOnlySimulation class as the microbe variant of this method is there.

Reply:
One more thought on this I just had: I have a slight recollection that I probably already made the world wait automatically if there are pending membranes, as that's probably needed for microbes. So this might not actually be a place that needs additional safety checks, but I still would like to see this code be moved so that it is easy to find the similar code.